### PR TITLE
[Google Blockly] Fix - Themes for read-only workspaces

### PR DESCRIPTION
**Pull request by mikeharv**
## Problem
AmyBW reported an issue where blocks in read-only workspaces were being rendered incorrectly. This is currently impacting all Google Blockly lab levels with blocks embedded in hints/instructions as well as our internal block pools tool on levelbuilder. [[Slack thread](https://codedotorg.slack.com/archives/C03DBDN67B7/p1678479415540099)]

Screen Shot 2023-03-10 at 12 15 32 PM

## Explanation
Typically, when we create a Blockly workspace, we call Blockly's `inject` method, but we skip this step for read-only workspaces. Blockly does support calling `new WorkspaceSvg() directly but we need to make sure it's doing everything correctly.

Attempting to manually set the theme in the dev console produced an error: 
image

`setTheme` calls `workspace.getInjectionDiv` which calls every element's `parentNode.getAttribute('class')` looking for `injectionDiv`:
https://github.com/google/blockly/blob/17064a1c396d392736b10fcd21493d262d98e27c/core/workspace_svg.ts#L702
Our read-only workspaces had no element with that class so Blockly couldn't determine the parent div. Additionally, because we were skipping the inject call it seems like we weren't properly applying the new theme to all elements.

## Solution

`injectionDiv` needs to be added to the list of classes in the svg we create for the workspace to fix this. We also need to add a call to `setTheme` before return the new workspace.

**Fixed:**
image

While we're at it, we can also refactor how the default theme is chosen for both standard and read-only workspaces. Because we want alignment in how the theme is chosen, this logic can be moved to a helper function. This allows newly created read-only workspaces to use the theme currently indicated by the browser's localStorage, as set by the workspace context menu.

**Read-workspaces rendering with the high contrast theme on page reload:**
image
image
 
While not related to the scope of this task, I also snuck in a fix to the CdoDark theme after @breville pointed out that it was missing the Gotham font specification.

## Follow-up work

Dynamically switching themes with the context menu only affects the main workspace (ie. the workspace within the scope of the context menu). We can update the logic in `contextMenu.js` to affect each of `Blockly.Workspace.getAll()` instead of just `scope.workspace`. This will make it so that switching themes will affect every workspace on screen at once. (https://codedotorg.atlassian.net/browse/SL-649 - Assigned to @bencodeorg)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/addons/cdoUtils.js [hunk 1]
```diff
@@ -91,3 +92,13 @@ export function getField(type) {
...
   return field;
 }
+
+/**
+ * Returns a theme object, based on the presence of an option in the browser's localStorage.
+ * @param {string} type
+ * @returns {?Blockly.Field}
+ */
+// Users can change their active theme using the context menu. Use this setting, if present.
+export function chooseTheme(themeOption) {
```
Comment (molly-moen) [inline apps/src/blockly/addons/cdoUtils.js]: nit: would a better name for this function be `getTheme`?

**Review by molly-moen — APPROVED** [review]
LGTM!

File: apps/src/blockly/addons/cdoUtils.js (hunk 1 above)
Comment (mikeharv) [reply apps/src/blockly/addons/cdoUtils.js]: `getTheme` is a core Blockly function that returns the current theme for a given workspace. The purpose of this function is to determine which theme should be _set_. Open to other suggestions!

Comment (molly-moen) [reply apps/src/blockly/addons/cdoUtils.js]: gotcha. Maybe something like `getChosenTheme` or `getActiveTheme`? `chooseTheme` to me sounds like the function is making a choice, but it's really reading the choice of the user, right?

Comment (mikeharv) [reply apps/src/blockly/addons/cdoUtils.js]: Yes, that's right. How about `getUserTheme`? We have an upcoming ticket to move this preference from localStorage to user preferences, and that might work well for when we get there. If we wanted it to be super accurate for now I could also see `getLocalStorageTheme` or something.

Comment (molly-moen) [reply apps/src/blockly/addons/cdoUtils.js]: `getUserTheme` sounds good!

**Review by bencodeorg — APPROVED** [review]

**Pull request merged**